comitup/config.py

```diff
@@ -43,29 +54,82 @@ def __getattr__(self, tag):
             raise AttributeError
 
 
+def _getserial():
+    # Extract serial from cpuinfo file
+    cpuserial = "0000000000000000"
+    try:
+        with open('/proc/cpuinfo', 'r') as f:
+            for line in f:
+                if line.startswith('Serial'):  # Serial		: 10000000082acda8
+                    cpuserial = line.split(':')[-1].strip()
+                    break
+
+    except Exception as e:
+        log.error("Failed to parse /proc/cpuinfo for serial number")
+        log.error(e)
+        cpuserial = "ERROR000000000"
+
+    return cpuserial
+
+
+def _get_mac():
+    mac = "000000000000"
+    cmd = "nmcli -f GENERAL.HWADDR device show wlan0".split(' ')
```

Comment:
You've hard-coded wlan0. It's been a while since I looked at packaging - is the WiFi MAC printed on the box?

Reply:
Yes.  Picked wlan0 over eth0.
Adafruit package does not have any MACs or serial numbers.  
My product does have a label, where I print both wlan0 and eth0 MACs and the RPi serial number.  I assume that IT needs the MACs to whitelist the device.